src/ebhack/MapEditor.java

```diff
@@ -2144,19 +2142,23 @@ private void importNPCs(File f) {
 
 		private void importSpriteGroups(Project proj) {
 			int w, h, x, y, z;
-			for (int i = 0; i < spriteGroups.length; ++i) {
-				spriteGroups[i] = new Image[4];
+			int i = 0;
+			while (true) {
+				spriteGroups.add(new Image[4]);
 				try {
-					BufferedImage sheet = ImageIO.read(new File(proj
-							.getFilename(
-									"eb.SpriteGroupModule",
-									"SpriteGroups/"
-											+ ToolModule.addZeros(i + "", 3))));
+					// TODO: this was using proj.getFilename. Why does that even exist?
+					String filename = proj.getDirectory() + File.separator + "SpriteGroups" + File.separator
+							+ ToolModule.addZeros(i + "", 3) + ".png";
+					File file = new File(filename);
+					if (!file.exists()) {
+						break;
+					}
```

Comment:
I have an idea for improving this.. I'll take a crack at it. Just installing IDEA now